src/components/Sticker/index.tsx

```diff
@@ -4,39 +4,80 @@ import classNames from 'classnames'
 
 import styles from './index.module.scss'
 
+type StyleType =
+  | 'neutral'
+  | 'secondary'
+  | 'tertiary'
+  | 'success'
+  | 'warning'
+  | 'error'
+
+type DotColor = 'success' | 'error' | 'warning' | 'neutral' | 'tertiary'
+
+type StyleCategory =
+  | { variant: 'withIcon'; icon: React.ReactNode }
+  | {
+      variant: 'withDot'
+      dotColor: DotColor
+      dotIcon?: React.ReactNode
```

Comment:
do we need to receive the `dotIcon` as a react node? is it supposed to be this general?